Lib/glyphsLib/builder/anchors.py

```diff
@@ -95,18 +113,36 @@ def _get_anchor_data(anchor_data, ufo, components, anchor_name):
         anchor_data[anchor.name] = t.transformPoint((anchor.x, anchor.y))
 
 
-def _adjust_anchors(anchor_data, ufo, component):
-    """Adjust anchors to which a mark component may have been attached."""
+def _componentAnchorFromLib(_glyph, _targetComponent):
+    """Pull component’s named anchor from Glyph.lib"""
+    if COMPONENT_INFO_KEY in _glyph.lib:
+        for _anchorLib in _glyph.lib[COMPONENT_INFO_KEY]:
+            if (
+                "anchor" in _anchorLib
+                and "name" in _anchorLib
+                and _anchorLib["name"] == _targetComponent.baseGlyph
+                and _anchorLib["index"] == _glyph.components.index(_targetComponent)
+            ):
+                return _anchorLib["anchor"]
 
+
+def _adjust_anchors(anchor_data, ufo, parent, component):
+    """Adjust anchors to which a mark component may have been attached."""
     glyph = ufo[component.baseGlyph]
     t = Transform(*component.transformation)
+    _componentAnchor = _componentAnchorFromLib(parent, component)
     for anchor in glyph.anchors:
-        # only adjust if this anchor has data and the component also contains
-        # the associated mark anchor (e.g. "_top" for "top")
+        # adjust either if this anchor has data and the component also contains
+        # the associated mark anchor (e.g. "_top" for "top") ...
         if anchor.name in anchor_data and any(
             a.name == "_" + anchor.name for a in glyph.anchors
         ):
             anchor_data[anchor.name] = t.transformPoint((anchor.x, anchor.y))
+        # ... or component is attached to a specific named anchor
+        # (e.g. top_2 for a ligature glyph)
+        # rather than to the standard anchors (top/bottom)
+        elif _componentAnchor in anchor_data:
```

Comment:
shouldn't the second case (ie. explicit named component.anchor whatever that is) take precedence over the first one?

Reply:
You are right. Earlier I had it flipped already but something didn't work, so I had left it untouched.
I just changed it around.